src/drt/src/gc/FlexGC_main.cpp

```diff
@@ -1692,6 +1649,181 @@ void FlexGCWorker::Impl::checkMetalShape_minEnclosedArea(gcPin* pin)
   }
 }
 
+void FlexGCWorker::Impl::checkMetalShape_lef58Area(gcPin* pin)
+{
+  if (ignoreMinArea_ || !targetNet_) {
+    return;
+  }
+
+  auto poly = pin->getPolygon();
+  auto layer_idx = poly->getLayerNum();
+  auto layer = getTech()->getLayer(layer_idx);
+
+  if (!layer->hasLef58AreaConstraint()) {
+    return;
+  }
+
+  auto constraints = layer->getLef58AreaConstraints();
+
+  auto sort_cmp =
+      [this](const frLef58AreaConstraint* a, const frLef58AreaConstraint* b) {
+        auto a_rule = a->getODBRule();
+        auto b_rule = b->getODBRule();
+
+        return a_rule->getRectWidth() < b_rule->getRectWidth();
+      };
+
+  // sort constraints to ensure the smallest rect width will have
+  // preference above other rect width statements
+  std::sort(constraints.begin(), constraints.end(), sort_cmp);
+
+  bool check_rect_width = true;
+
+  for (auto con : constraints) {
+    odb::dbTechLayerAreaRule* db_rule = con->getODBRule();
+    auto min_area = db_rule->getArea();
+    auto curr_area = gtl::area(*poly);
+
+    if (curr_area >= min_area) {
+      continue;
+    }
+
+    gtl::rectangle_data<frCoord> bbox;
+    gtl::extents(bbox, *pin->getPolygon());
+    Rect bbox2(gtl::xl(bbox), gtl::yl(bbox), gtl::xh(bbox), gtl::yh(bbox));
+    if (!drWorker_->getDrcBox().contains(bbox2))
+      continue;
+    for (auto& edges : pin->getPolygonEdges()) {
+      for (auto& edge : edges) {
+        if (edge->isFixed())
+          continue;
+      }
+    }
+
+    // add patch only when the poly is not a rect
+    if (checkMetalShape_lef58Area_exceptRectangle(poly, db_rule)) {
+      checkMetalShape_addPatch(pin, min_area);
+    }
+
+    // add patch only if poly is rect and its width is less than or equal to
+    // rectWidth value on constraint
+    if (check_rect_width
```

Comment:
What happens if someone defines a LEF58_AREA without setting RECTWIDTH or EXCEPTRECTANGLE?  Is it still checked?

Reply:
I have a question about this. My interpretation of the lefdef manual is that if we have a LEF58_AREA with RECTWIDTH or with EXCEPTRECTANGLE, we'll not have anything else in this specific property. Is this correct?

E.g.: this property declaration would be invalid:
PROPERTY LEF58_AREA
       "AREA 0.08 RECTWIDTH 0.05 MASK 2 ;"

In any case, the curr code is incorrect, I'll push a temp fix to call the patch when only the area is defined.

<img width="1106" alt="image" src="https://user-images.githubusercontent.com/43450810/192389697-433243e0-3673-420a-ace8-e44888cb26f6.png">

Reply:
Btw, the min area value is always checked at the beginning of the function. With my latest commit, I'm now applying the patch when only AREA is defined.

Reply:
Knowing what the LEF manual means is always half the problem.  Usually the 58 properties are a superset of the older syntax.  I don't know what it means to have and '|' with a required field.  I suspect they just forgot the [].

I can say that in one of our private pdks I see 
```
  PROPERTY LEF58_AREA "
  AREA X.XXX ; " ;
```
Its not a layer we route on but it suggests that it is allowed.

Reply:
Yes, I think they forgot the `[]` too. My doubt is more related to the `|` symbols. Is it implying that we can only have one of the keywords? Like, is `AREA 0.8 RECTWIDTH 0.1 EXCEPTEDGELENGTH 0.5` valid? From what I understand, it is not valid.

Reply:
I don't think it makes sense to combine those two and I don't see any examples of it in our PDKs.  I could imagine combining EXCEPTRECTANGLE with LAYER so the semantics of the '|' is unclear.  Don't stress to much over weird cases and just give an error if they happen - we can always address them if they come up.

Reply:
Got it. I'm finishing testing with our PDKs and will let you know when I finish.